### ISSUE TITLE
Unexpected id selector (selector-no-id) SASS interpolation and SCSS Maps

### ISSUE THREAD

**Issue by alan-agius4** [stylelint/stylelint#829]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `&:nth-child(#{map-get($variant, position)})` is being flagged as an ID while it shouldn't. When the map-get is replaced with a SASS variable ex: `&:nth-child(#{$position})` it works fine.

> Which rule, if any, is this issue related to?

`selector-no-id`

> What CSS is needed to reproduce this issue?

```sass
@mixin betslip-coupon-col-variant($variant: ()){
	$variant-defaults: (
		position: 1,
		width: 100%
	);
	
	$variant: map-merge($variant-defaults, $variant);
	
	&:nth-child(#{map-get($variant, position)}) {
		width: map-get($variant, width);
	}
}
```

> What stylelint configuration is needed to reproduce this issue?
```json
{
  "rules": {
   "selector-no-id": true,
   "selector-list-comma-newline-after": "always",
  }
}
```

> Which version of stylelint are you using?

`4.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Gulp and PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps and SASS interpolation. 
I have looked a bit in your code and this might be an issue with `postcss-selector-parser`

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:
```
Unexpected id selector (selector-no-id)
Expected newline after "," (selector-list-comma-newline-after)
```
